src/backend/access/heap/heapam.c

```diff
@@ -8369,11 +8370,12 @@ log_heap_update(Relation reln, Buffer oldbuf,
 	xlrec.old_xmax = HeapTupleHeaderGetRawXmax(oldtup->t_data);
 	xlrec.old_infobits_set = compute_infobits(oldtup->t_data->t_infomask,
 											  oldtup->t_data->t_infomask2);
+	xlrec.t_cid = HeapTupleHeaderGetRawCommandId(oldtup->t_data);
 
 	/* Prepare WAL data for the new page */
 	xlrec.new_offnum = ItemPointerGetOffsetNumber(&newtup->t_self);
 	xlrec.new_xmax = HeapTupleHeaderGetRawXmax(newtup->t_data);
-	xlrec.t_cid = HeapTupleHeaderGetRawCommandId(newtup->t_data);
```

Comment:
Hmm, with this change, we don't store the new tuple's cmax in the WAL record anymore. The new tuple cannot be restored correctly then.